src/EnergyPlus/GlobalNames.cc

```diff
@@ -75,11 +75,11 @@ void IntraObjUniquenessCheck(EnergyPlusData &state,
     if (NameToVerify.empty()) {
         ShowSevereError(state, format("E+ object type {} cannot have a blank {} field", CurrentModuleObject, FieldName));
         ErrorsFound = true;
-        NameToVerify = "xxxxx";
+        // NameToVerify = "xxxxx";
         return;
     }
 
-    auto const &find_string = UniqueStrings.find(NameToVerify);
+    auto const find_string = UniqueStrings.find(NameToVerify);
```

Comment:
I need to go back through these branches and make a list of these. I'm never going to remember all of them.

Reply:
`grep "auto " *.cc */*.cc | grep -v "\->begin()\|\->end()\|\.begin()\|\.end()\|\.find(\|\.next(\|\.find_\|\.emplace(" | grep -v "auto &" | grep\
 -v "auto const &" | grep -v " = \[" | grep -v "auto \*" | grep -v "auto const \*" | grep -v "AUTO_OK" | grep -v ".readLine()\|readFile(\|read\
JSON(\|\.open(\|\.try_open(" | grep -v "std::next\|std::find" | grep -v "std::make_shared\|std::make_unique" | grep -v "auto \["`

Reply:
I also just came up with a mnemonic! BENEFIT. If it's returned by Begin End Next Emplace or Find, it's an ITerator and needs plain `auto`.